src/parser.ts

```diff
@@ -112,10 +125,12 @@ export function byNode<T extends ParserObject>(node: HTMLElement): T {
 /**
  * The public API for retrieving a parser object by ID
  * @param  {string}                 id The ID of the parser object to retrieve
+ * @param  {Document}               doc The optional Document. Defaults to `document`.
  * @return {T extends ParserObject}    The instance associated with the ID or `undefiend`
  */
-export function byId<T extends ParserObject>(id: string): T {
-	return <T> idMap[id];
+export function byId<T extends ParserObject>(id: string, doc: Document = document): T {
```

Comment:
If it doesn't bother you that the `id` argument would no longer be in the same position in each context, then I would be okay with such an update.